src/Databases/DatabaseOrdinary.cpp

```diff
@@ -323,23 +323,42 @@ void DatabaseOrdinary::loadTableFromMetadata(
 
     LOG_TRACE(log, "Loading table {}", name.getFullName());
 
-    try
-    {
-        auto [table_name, table] = createTableFromAST(
-            query,
-            name.database,
-            getTableDataPath(query),
-            local_context,
-            mode);
-
-        attachTable(local_context, table_name, table, getTableDataPath(query));
-    }
-    catch (Exception & e)
+    constexpr size_t max_tries = 3;
+    size_t tries = 0;
+    time_t sleep_time = 1;
+
+    while (true)
     {
-        e.addMessage(
-            "Cannot attach table " + backQuote(name.database) + "." + backQuote(query.getTable()) + " from metadata file " + file_path
-            + " from query " + serializeAST(query));
-        throw;
+        try
+        {
+            auto [table_name, table] = createTableFromAST(
+                query,
+                name.database,
+                getTableDataPath(query),
+                local_context,
+                mode);
+
+            attachTable(local_context, table_name, table, getTableDataPath(query));
+            return;
+        }
+        catch (Coordination::Exception & e)
+        {
+            if (!Coordination::isHardwareError(e.code))
+                throw;
+            tryLogCurrentException(log);
+            sleepForSeconds(sleep_time);
+            sleep_time *= 2;
+            ++tries;
+            if (tries > max_tries)
+                throw;
```

Comment:
now we don't append the message with `Cannot attach table...` in case of Coordination exceptions